src/stan/callbacks/chained_writer.hpp

```diff
@@ -0,0 +1,89 @@
+#ifndef STAN_CALLBACKS_CHAINED_WRITER_HPP
+#define STAN_CALLBACKS_CHAINED_WRITER_HPP
+
+#include <stan/callbacks/writer.hpp>
+#include <ostream>
+#include <vector>
+#include <string>
+
+namespace stan {
+  namespace callbacks {
+
+    /**
+     * <code>chained_writer</code> is an implementation that writes to
+     * multiple writers.
+     *
+     * For any given call to this writer, it will delegate to both writers
+     * provided in the constructor.
+     */
+    class chained_writer : public writer {
+    public:
+      /**
+       * Constructor.
+       *
+       * @param[in, out] writer1 first writer
+       * @param[in, out] writer2 second writer
+       */
+      chained_writer(writer& writer1,
+                     writer& writer2)
+        : writer1_(writer1), writer2_(writer2) {
+      }
+
+
+      void operator()(const std::string& key, double value) {
+        writer1_(key, value);
+        writer2_(key, value);
+      }
+
+      void operator()(const std::string& key, int value) {
+        writer1_(key, value);
+        writer2_(key, value);
+      }
+
+      void operator()(const std::string& key, const std::string& value) {
+        writer1_(key, value);
+        writer2_(key, value);
+      }
+
+      void operator()(const std::string& key,
+                      const double* values,
```

Comment:
The doc is inherited in doxygen.
The reason for the pointer was to keep in line with the old pull request. I really think IO should be handled better, so I'm going to scale back the writers to what we use now:
- `writer()`
- `writer(std::string)`
- `writer(std::vector<std::string>)`
- `writer(std::vector<double>)`

It'll cut down on our overhead for now. I think it's the right way to go and we won't have pointers floating around.